cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -33,6 +34,31 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// testOpts represents parameters required for a single unit test
+type testOpts struct {
+	description string
+	pods        []*apiv1.Pod
+	pdbs        []*policyv1.PodDisruptionBudget
+	rcs         []*apiv1.ReplicationController
+	replicaSets []*appsv1.ReplicaSet
+	expectFatal bool
+	// expectFatalFn is ORed with expectFatal to decide fatality
+	// i.e., if (expectFatal || expectFatalFn()) = true, fatal
+	// if !(expectFatal || expectFatalFn()) = false , not fatal
+	expectFatalFn func() bool
+	expectPods    []*apiv1.Pod
+	// expectPodsFn is to provide more flexibility in adding pods
+	// returned pods are appended to expectPods above
+	expectPodsFn        func() []*apiv1.Pod
+	expectDaemonSetPods []*apiv1.Pod
+	expectBlockingPod   *BlockingPod
+	// expectBlockingPodFn provides more flexibility than expectBlockingPod
+	// result from this function is used only if expectBlockingPod is nil
+	expectBlockingPodFn func() *BlockingPod
+}
+
+var skipNodesWithCustomControllerPods bool = false
```

Comment:
I get that this helps reduce # of lines of code, but making unit tests rely on injected functions that use a global variable is tricky. I think it would be best to keep unit tests simple (even if overly verbose), to make them easier to reason about. Otherwise we would have to have tests of the tests to check whether they work correctly.

Also, introducing a global prevents us from ever running the test cases in parallel.

Reply:
I see your point about not making tests complicated. Let me think about this. 

> Also, introducing a global prevents us from ever running the test cases in parallel.

I am a little on the fence about this but I am not too happy about using global variable either. Let me think more about this.

Reply:
Updated the code (removed global variable and injected functions). Might need a little more feedback.